### PR TITLE
Run static analysis on mazerunner scenarios

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

[Detekt](https://github.com/detekt/detekt), [Ktlint](https://github.com/pinterest/ktlint), and checkstyle don't currently run on the mazerunner scenario code. This changeset ensures that the tools run against mazerunner scenarios to enforce code quality, and resolves existing style violations.

## Changeset

- Setup ktlint, detekt, and checkstyle to run against mazerunner
- Added a separate CI job that lints mazerunner code
- Resolved existing style violations (each tool has a separate commit in the history)

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -135,6 +135,14 @@ steps:
           run: android-jvm
     command: 'bash ./scripts/run-cpp-check.sh'
 
+  - label: ':android: Lint mazerunner scenarios'
+    depends_on: "android-jvm"
```
Comment (twometresteve) [inline .buildkite/pipeline.yml:139]: Will this need to change after the docker image rejig?

File: features/fixtures/mazerunner/cxx-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/CXXThrowSomethingOutsideReleaseStagesScenario.java [hunk 2]
```diff
@@ -17,7 +15,11 @@
 
     public native void crash(int num);
 
-    public CXXThrowSomethingOutsideReleaseStagesScenario(@NonNull Configuration config, @NonNull Context context) {
+    /**
```
Comment (twometresteve) [inline features/fixtures/mazerunner/cxx-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/CXXThrowSomethingOutsideReleaseStagesScenario.java:18]: Do we really want an empty JavaDoc?

File: features/fixtures/mazerunner/jvm-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/NewSessionScenario.kt [hunk 3]
```diff
@@ -28,30 +30,32 @@ internal class NewSessionScenario(config: Configuration,
...
+                client.startSession()
+                Log.d("Bugsnag - New", "First session started")
+                Thread.sleep(SLEEP_MS)
+                flushAllSessions()
+                Log.d("Bugsnag - New", "First session flushed")
+                Thread.sleep(SLEEP_MS)
+                client.notify(generateException())
+                Log.d("Bugsnag - New", "First exception notified")
+                Thread.sleep(SLEEP_MS)
+// stop tracking the existing session
```
Comment (twometresteve) [inline features/fixtures/mazerunner/jvm-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/NewSessionScenario.kt:45]: Surprised that comments with no indentation are desirable/allowed?

**Review by twometresteve — APPROVED** [review]
A few questions asked, but this is great to see.  I'll be sleeping more peacefully at night.

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (fractalwrench) [reply .buildkite/pipeline.yml:139]: Yes - this depends on which PR lands first.

File: features/fixtures/mazerunner/jvm-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/NewSessionScenario.kt (hunk 3 above)
Comment (fractalwrench) [reply features/fixtures/mazerunner/jvm-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/NewSessionScenario.kt:45]: I don't believe there are indentation rules for detekt/ktlint by default, so have amended the indentation here manually.

File: features/fixtures/mazerunner/cxx-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/CXXThrowSomethingOutsideReleaseStagesScenario.java (hunk 2 above)
Comment (fractalwrench) [reply features/fixtures/mazerunner/cxx-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/CXXThrowSomethingOutsideReleaseStagesScenario.java:18]: Added comment to this Javadoc - this is to suppress a checkstyle rule that public methods over 1 line long must have a comment.

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (fractalwrench) [reply .buildkite/pipeline.yml]: Switched target to the `docker-reorg` branch as that will land first.

**Pull request merged**